### PR TITLE
KafkaSinkCluster: scram_over_mtls broker discovery

### PR DESCRIPTION
**Pull request by rukai**
progress towards https://github.com/shotover/shotover-proxy/issues/1618

This PR:
* introduces a new integration test `cluster_sasl_scram_over_mtls_multi_shotover`, previously we just had `cluster_sasl_scram_over_mtls_single_shotover`
   + This tests with multiple shotover's running which could reveal some edge cases
   + It also uses various numbers of elements in `mtls_port_contact_points` for each shotover yaml, which will exercise different cases within the broker discovery logic
* Replaces the simple list of connections with a list of nodes.
   + This is needed so we can avoid creating duplicate connections and recreate connections if they go down.
* Adds a `find_new_brokers` method which performs broker discovery.

### CODE REVIEW

File: shotover-proxy/tests/test-configs/kafka/cluster-sasl-scram-over-mtls/topology3.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                rack: "rack0"
+                broker_id: 1
+              - address: "127.0.0.1:9193"
+                rack: "rack0"
+                broker_id: 2
+            local_shotover_broker_id: 0
+            # every shotover node purposefully tests a different number of contact points
+            first_contact_points: ["172.16.1.2:9092", "172.16.1.3:9092", "172.16.1.4:9092"]
+            authorize_scram_over_mtls:
+              mtls_port_contact_points: ["172.16.1.2:9094"]
```
Comment (rukai) [inline shotover-proxy/tests/test-configs/kafka/cluster-sasl-scram-over-mtls/topology3.yaml:22]: move the variations from first_contact_points to mtls_port_contact_points

**Review by ronycsdu — APPROVED** [review]

**Review by justinweng-instaclustr — APPROVED** [review]
